offsets_db_api/routers/charts.py

```diff
@@ -860,21 +874,43 @@ async def get_credits_by_category(
         ('retired', retired_max, '<=', Project),
     ]
 
-    for attribute, values, operation, model in filters:
-        query = apply_filters(
-            statement=query, model=model, attribute=attribute, values=values, operation=operation
-        )
-
     # Handle 'search' filter separately due to its unique logic
     if search:
         search_pattern = f'%{search}%'
-        query = query.where(
+        matching_projects = matching_projects.where(
             or_(
                 col(Project.project_id).ilike(search_pattern),
                 col(Project.name).ilike(search_pattern),
             )
         )
 
+    if beneficiary_search:
```

Comment:
I have no idea how possible/worthwhile it is to do this refactor, but it seems like we now have similar logic duplicated across three endpoints (`/charts/credits_by_category`, `/credits`, `/projects`) that could potentially be shared. It seems like defining filtering logic in just once place would make it easier to ensure consistency if/when future changes arise (e.g., filtering based on @badgley's harmonized fields).

Reply:
👍🏽 this is a great idea